### PR TITLE
Add ability to compile script to use non-intel compilers; add Cheyenne modulefiles for gnu and intel

### LINKED ISSUE #449: No Cheyenne modulefile for compile_upp.sh
There is no modulefile for using the compile_upp.sh script on Cheyenne.

I am working on a PR to add this, but I do not have permissions to assign myself.

### PR DESCRIPTION
**Pull request by mkavulich**
This PR enhances the compile_upp.sh script to allow for the "-c COMPILER" argument, as well as updating the usage statement, and adding a print statement that states which machine and compiler we are compiling for. This requires renaming existing modulefiles to the naming scheme ${machine}_intel (as the existing machines all use intel compilers, which remains the default option). Additionally, I have added a verbose exit statement for the case where the selected modulefile does not exist.

This PR also adds `cheyenne_intel` and `cheyenne_gnu` modulefiles, so UPP can now be built with the "compile_upp.sh" script on Cheyenne for either supported compiler. This was tested by running the SRW App with several end-to-end cases for both Intel and Gnu compilers (resulting executable was copied in manually since the App does not use this compile script); all UPP tasks were successful.

Behavior on platforms other than Cheyenne should not change.

### CODE REVIEW

**Comment by WenMeng-NOAA** [comment]
@mkavulich The modulefile name change breaks UPP building on WCOSS2. Would you try the name "wcoss2_intel.lua"?

**Comment by WenMeng-NOAA** [comment]
@mkavulich Did you test UPP building for all platforms with modulefile naming changes? The easy way might be adding
compiler name(e.g. cheyenne_gnu) in the modulefiles for non-intel compiler option.

File: tests/compile_upp.sh [hunk 1]
```diff
@@ -60,7 +64,17 @@ if [[ $MACHINE_ID != "unknown" ]]; then
       module purge
    fi
    module use $PATHTR/modulefiles
-   modulefile=${MACHINE_ID}
+   if [[ $compiler == "intel" ]]; then
+      modulefile=${MACHINE_ID}
+   else
+      modulefile=${MACHINE_ID}_${compiler}
+   fi
+   if [ -f "${PATHTR}/modulefiles/${modulefile}" ]; then
```
Comment (WenMeng-NOAA) [inline tests/compile_upp.sh]: @mkavulich On WCOSS2, we use modulefile in lua format. You might add changes as
if [ -f "${PATHTR}/modulefiles/${modulefile}" -o \
     -f "${PATHTR}/modulefiles/${modulefile}.lua" ]; then

**Comment by WenMeng-NOAA** [comment]
@mkavulich Now the changes look good to me. Can you open an UPP issue linking this PR?

File: tests/compile_upp.sh (hunk 1 above)
Comment (mkavulich) [reply tests/compile_upp.sh]: @WenMeng-NOAA Good catch, I have pushed this change. I have tested the build and running on Cheyenne (intel and gnu), and the build on Hera, Jet, and Orion.  I will need someone else to test other platforms as I do not have access.

Comment (WenMeng-NOAA) [reply tests/compile_upp.sh]: @mkavulich  I will test on NOAA R&D machines, e.g. Hera, Orion.

**Comment by WenMeng-NOAA** [comment]
The UPP regression tests were completed on WCOSS1-Dell and Hera. No changed results were found. The build only tests also 
were completed on Orion. @mkavulich This PR is waiting for a relevant UPP issue  (from https://github.com/NOAA-EMC/UPP/issues) for merging.

**Comment by WenMeng-NOAA** [comment]
@mkavulich Found the issue #449 for this PR. The PR is ready for merging. Thanks!

**Review by WenMeng-NOAA — APPROVED** [review]

**Comment by edwardhartnett** [comment]
Shouldn't this all be done in the CMake build, and no build script required?

**Comment by WenMeng-NOAA** [comment]
Yes, the build script is tests/compile_upp.sh.

**Pull request merged**